tests/v2/test_wakunode.nim

```diff
@@ -144,3 +144,40 @@ procSuite "WakuNode":
       (await completionFut.withTimeout(5.seconds)) == true
     await node1.stop()
     await node2.stop()
+
+  asyncTest "Filter protocol returns expected message":
+    let
+      nodeKey1 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node1 = WakuNode.init(nodeKey1, ValidIpAddress.init("0.0.0.0"),
+        Port(60000))
+      nodeKey2 = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node2 = WakuNode.init(nodeKey2, ValidIpAddress.init("0.0.0.0"),
+        Port(60002))
+      contentTopic = "foobar"
+      message = WakuMessage(payload: "hello world".toBytes(), contentTopic: contentTopic)
+
+    var completionFut = newFuture[bool]()
+
+    await node1.start()
+    await node2.start()
+
+    node1.wakuFilter.setPeer(node2.peerInfo)
+
+    proc filterHandler(msg: MessagePush) {.gcsafe, closure.} =
+      echo msg
+      check:
+        msg.messages[0] == message
+      completionFut.complete(true)
+
+    await node1.filter(FilterRequest(topic: "waku", contentFilter: @[ContentFilter(topics: @[contentTopic])]), filterHandler)
+
+    await sleepAsync(2000.millis)
+
+    await node2.subscriptions.notify("waku", message)
```

Comment:
What about a test where this is called? I.e. it receives messages from a third node e.g.

In addition to that, a third test would be useful to test the negative case. I.e. if you have three nodes A B C and C sends stuff to B, then A MUST NOT receive traffic from C on a different content topic. This could possibly implied by the absence of notify, but when both nodes speak relay (for publishing) and filter, it could perhaps be forwarded in some way.

Reply:
@oskarth yeah that sounds good. Should we do something similar for the store protocol. Can hihglight in an issue.

Reply:
Ok, please paste link here once done.